source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -129,6 +130,23 @@ void Filter::onGrpcClose() {
   }
 }
 
+void Filter::sendImmediateResponse(const ImmediateResponse& response) {
+  const auto status_code = response.has_status() ? response.status().code() : 200;
+  const auto grpc_status =
+      response.has_grpc_status()
```

Comment:
Is this behavior covered in the tests below?

Reply:
Line coverage was OK but I think the coverage of all the cases could be better -- fixed that in the unit test.